daemon-scheduler/internal/features/wrappers/autoscaling_wrapper.go

```diff
@@ -44,3 +51,115 @@ func (wrapper AutoScalingWrapper) SetDesiredCapacity(asg string, count int64) er
 	}
 	return nil
 }
+
+func (wrapper AutoScalingWrapper) CreateAutoScalingGroup(asg string, configName string, availabilityZones []*string) error {
+	minSize := int64(0)
+	// MaxSize is set to 10 because it's the maximum number of instances launched by the e2e test
+	maxSize := int64(10)
+	in := autoscaling.CreateAutoScalingGroupInput{
+		AutoScalingGroupName: aws.String(asg),
+		MinSize: aws.Int64(minSize),
+		MaxSize: aws.Int64(maxSize),
+		LaunchConfigurationName: aws.String(configName),
+		AvailabilityZones: availabilityZones,
+	}
+
+	_, err := wrapper.client.CreateAutoScalingGroup(&in)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (wrapper AutoScalingWrapper) GetAutoScalingGroupStatus(asg string) (string, error) {
+	in := autoscaling.DescribeAutoScalingGroupsInput{
+		AutoScalingGroupNames: []*string{aws.String(asg)},
+	}
+
+	resp, err := wrapper.client.DescribeAutoScalingGroups(&in)
+	if err != nil {
+		return "", err
+	}
+
+	var status string
+	if resp.AutoScalingGroups[0].Status != nil {
```

Comment:
What will happen if `resp.AutoScalingGroups` has a length of 0? My guess is it will nil pointer panic here.